### ISSUE TITLE
Message Timestamp

### ISSUE THREAD

**Issue by fpoltronieri** [streadway/amqp#240]
Hello,
I have noticed that when I received a message, I am not able to retrieve the correct timestamp in nanoseconds.
This happens because the received Timestamp has the time.Time.nsec field set to 0.
Looking at the AMQP specification the timestamp is an int64, and a go time.Time type requires at least 96 bit (64 bit for the number of seconds and 32 for the number of nanosecond).

**Comment by michaelklishin** [comment]
Correct, AMQP 0-9-1 assumes the timestamp property is in seconds. You can pass a larger integer value in a header.

**Comment by fpoltronieri** [comment]
>  You can pass a larger integer value in a header.
Does it means that I can pass a larger value in the body of the AMQP message?

**Comment by michaelklishin** [comment]
You can pass anything in message bodies. For this client and RabbitMQ, they are just bags of bytes ;)

**Comment by fpoltronieri** [comment]
Thanks :)

**Comment by michaelklishin** [comment]
OK, so we concluded this warrants a documentation update. Converting `time.Time` to seconds should be straightforward enough.

**Comment by terrywh** [comment]
but using another consumer driver may result in inconsistency:

![image](https://user-images.githubusercontent.com/968222/43771283-4eba5a60-9a71-11e8-985e-2a91482b2d7c.png)

as the timestamp should be in seconds
